public_html/composer.json

```diff
@@ -0,0 +1,129 @@
+{
+	"name": "wordcamp/wordcamp.org",
+	"description": "",
+	"homepage": "https://wordcamp.org",
+	"license": "GPL-2.0-or-later",
+	"support": {
+		"issues": "https://github.com/WordPress/wordcamp.org/issues"
+	},
+	"config": {
+		"platform": {
+			"php": "7.2"
+		},
+		"vendor-dir": "wp-content/mu-plugins/vendor"
+	},
+	"repositories": [
+		{
+			"type": "composer",
+			"url": "https://wpackagist.org/"
+		},
+		{
+			"type": "package",
+			"package": [
+				{
+					"name": "wordpress-plugin/bbpress",
+					"type": "wordpress-plugin",
+					"version": "2.6",
+					"source": {
+						"type": "svn",
+						"url": "https://plugins.svn.wordpress.org/bbpress/",
+						"reference": "branches/2.6/"
```

Comment:
> Right now it just uses wp-cli to install the latest version of everything via a bash script.

- [ ] `SHOULD` - Since committers will have the SVN externals checked out as well, it seems like everyone else using  the latest version is good enough. Prod usually runs the latest anyway, so if we can setup Composer to pull the latest -- IIRC there's a `latest.zip` redirect available -- then that'd save us a bunch of extra work when updating

- [ ] `SHOULD` - If we go with that approach, it'd be good to document the reason. Not sure where it'd be best, though, b/c JSON maddeningly doesn't support comments (b/c in theory it's just for data, but obviously that's not how it's used in reality). 
    - [ ] `MIGHT` - Maybe composer supports alternative formats like YAML (which has it's own annoyances, but at least allows comments)?

Reply:
Noting that as far as I can tell, composer does not support alternative formats like yml for its package files.

Reply:
I'll document the reason for putting plugins/themes in dev dependencies in the merge commit. There doesn't seem like any other good place for it at the moment, given the json limitations you mention.

Reply:
I worry that it'll get lost there. I wonder if the the `svn:externals` property in production's `wp-content/plugins` folder would be a good place? If someone ever mistakenly thought that the externals were no longer needed, then they hopefully see it when they went to delete the property?

Another good place might be in our private devdocs, similar to how we mention other security considerations around provisioning local and remove sandboxes?

Reply:
Private devdocs updated.

Reply:
What do you think about using `latest.zip` so that we don't have to manually update `composer.json` all the time? The only people using it for `plugins/themes` will be non-committers, and they don't need to worry about whether the version number is a few days/weeks out of date.

Reply:
If we're going to use **latest.zip**, it seems like we might as well switch back to the bash script. In my mind, the only advantage of putting these in composer.json is being able to specify version numbers, so if we're not doing that, the hassles and security concerns might not be worth it.

Reply:
🤔 I thought that you wanted to define them in `composer.json` so that all 3rd-party dependencies were managed in one place?

It seems like the advantage over the Bash script is that regular contributors only need to run `composer install`, instead of `sh install-plugins-themes.sh && composer install`. If we already need to use composer for the Meetup client, we might as well use it for other stuff too? 

Maybe I misunderstood that part of it, though? I guess I don't feel strongly either way. If we do stick w/ composer, though, I'd much rather pin to the latest, rather than having to constantly update version numbers in 4 places instead of 1.

Reply:
Moved to https://github.com/WordPress/wordcamp.org/issues/159